tools/third_party.py

```diff
@@ -200,10 +201,46 @@ def download_from_google_storage(item, bucket):
     ],
         env=google_env())
 
+# Download the given item from Chrome Infrastructure Package Deployment.
+def download_from_cipd(item):
+    if sys.platform == 'win32':
+        root_dir = "v8/buildtools/win"
+        item += "windows-amd64"
+    elif sys.platform == 'darwin':
+        root_dir = "v8/buildtools/mac"
+        item += "mac-amd64"
+    elif sys.platform.startswith('linux'):
+        root_dir = "v8/buildtools/linux64"
+        item += "linux-amd64"
+
+    # init cipd if necessary
+    if not os.path.exists(path.join(tp(root_dir), ".cipd")):
+        run([
+            tp('depot_tools/cipd'),
+            'init',
+            tp(root_dir),
+            '-force',
+        ],
+            env=google_env())
+
+    # read gn CIPD version from the DEPS file
+    deps = open(tp('v8/DEPS'), 'rb').read()
+    version = re.search("'gn_version': '(git_revision:([0-9a-f]{40}))'", deps).group(1)
+
+    run([
+        tp('depot_tools/cipd'),
+        'install',
+        item,
+        version,
+        '-root',
+        tp(root_dir),
+    ],
+        env=google_env())
+
 
 # Download gn from Google storage.
 def download_gn():
-    download_from_google_storage('gn', 'chromium-gn')
+    download_from_cipd('gn/gn/')
 
 
 # Download clang-format from Google storage.
```

Comment:
I've pushed a commit that updates depot_tools in deno_thrid_party.

Could you also update `download_clang_format()` to use `download_from_cipd()` and remove the code for `download_from_google_storage()`?

Reply:
Unfortunately `clang_format` isn't available on CIPD (yet), so we will have to use both methods (at least for a while).